helix-view/src/document.rs

```diff
@@ -1385,12 +1427,33 @@ impl Document {
         &self.diagnostics
     }
 
-    pub fn set_diagnostics(&mut self, diagnostics: Vec<Diagnostic>) {
-        self.diagnostics = diagnostics;
+    pub fn shown_diagnostics(&self) -> impl Iterator<Item = &Diagnostic> {
```

Comment:
```suggestion
    pub fn shown_diagnostics(&self) -> impl Iterator<Item = &Diagnostic> +  DoubleEndedIterator {
```

This ensure you can do `rev` without collecting